azurerm/internal/services/containers/kubernetes_nodepool.go

```diff
@@ -368,7 +367,6 @@ func FlattenDefaultNodePool(input *[]containerservice.ManagedClusterAgentPoolPro
 			"name":                  name,
 			"node_count":            count,
 			"node_labels":           nodeLabels,
-			"node_taints":           nodeTaints,
```

Comment:
so this isn't a breaking change, for now we'll need to set an empty list into the state:

```suggestion
			"node_taints":           []string{},
```

(we can remove this once we remove the field in 3.0)